notes_app/notes/api.py

```diff
@@ -39,6 +39,44 @@ def delete_detail(self, object_list, bundle):
         raise Unauthorized("Deletes not allowed")
 
 
+# Custom User Authorization
+class UserObjectsOnlyAuthorization(Authorization):
+    def read_list(self, object_list, bundle):
+        # This assumes a ``QuerySet`` from ``ModelResource``.
+        return object_list.filter(user=bundle.request.user)
+
+    def read_detail(self, object_list, bundle):
+        # Is the requested object owned by the user?
+        return bundle.obj.user == bundle.request.user
+
+    def create_list(self, object_list, bundle):
+        # Assuming they're auto-assigned to ``user``.
+        return object_list
+
+    def create_detail(self, object_list, bundle):
+        return bundle.obj.user == bundle.request.user
```

Comment:
I'm not sure that it should be here. Imagine registration step - user will not be able to register since he isn't logged in  and absent in `bundle.request.user`